nixos/lib/testing-python.nix

```diff
@@ -26,14 +26,17 @@ in rec {
 
     nativeBuildInputs = [ makeWrapper ];
     buildInputs = [ (python3.withPackages (p: [ p.ptpython ])) ];
-    checkInputs = with python3Packages; [ pylint black ];
+    checkInputs = with python3Packages; [ pylint black mypy ];
 
     dontUnpack = true;
 
     preferLocalBuild = true;
 
     doCheck = true;
     checkPhase = ''
+      mypy --disallow-untyped-defs \
+           --no-implicit-optional \
+           --ignore-missing-imports ${testDriverScript}
```

Comment:
why is the ignore missing imports part necessary?

Reply:
Because third-party libraries are imported that do not have type annotations.

Reply:
Because third-party library are imported that do not have type annotations.